File: yurt-convention-api/src/main/java/com/github/liulus/yurt/convention/exception/ServiceValidException.java
package com.github.liulus.yurt.convention.exception;

import com.github.liulus.yurt.convention.data.ServiceCode;

/**
 * @author liulu
 * @version V1.0
 * @since 2020/11/9
 */
public class ServiceValidException extends ServiceRuntimeException {

    public ServiceValidException() {
        super();
    }

    protected ServiceValidException(String message) {
        super(message);
    }

    protected ServiceValidException(String message, Throwable cause) {
        super(message, cause);
    }

    public ServiceValidException(String code, String message) {
        super(code, message);
    }

    public ServiceValidException(String code, String message, Throwable cause) {
        super(code, message, cause);
    }

    public ServiceValidException(ServiceCode serviceCode) {
        super(serviceCode);
    }

    public ServiceValidException(ServiceCode serviceCode, Throwable cause) {
        super(serviceCode, cause);
    }
    
}

File: yurt-convention-api/src/main/kotlin/com/github/liulus/yurt/convention/exception/ServiceValidException.kt
package com.github.liulus.yurt.convention.exception

import com.github.liulus.yurt.convention.enum.CommonCode

/**
 *
 * @author liulu
 * @version V1.0
 * @since 2020/10/31
 */
class ServiceValidException : ServiceRuntimeException {

    constructor() : super()

    constructor(message: String) : super(CommonCode.INVALID_PARAM.code(), message)

    constructor(message: String, cause: Throwable) : super(CommonCode.INVALID_PARAM.code(), message, cause)

    constructor(cause: Throwable) : super(cause)

    constructor(code: String, message: String) : super(code, message)

    constructor(code: String, message: String, cause: Throwable) : super(code, message, cause)


}

